coramin/third_party/minlplib_tools.py

```diff
@@ -0,0 +1,336 @@
+            continue
+
+        nquadcons = int(row[headings['nquadcons']])
+        if nquadcons < min_nquadcons or nquadcons > max_nquadcons:
+            logger.debug('excluding {case} due to nquadcons'.format(case=case_name))
+            continue
+
+        npolynomcons = int(row[headings['npolynomcons']])
+        if npolynomcons < min_npolynomcons or npolynomcons > max_npolynomcons:
+            logger.debug('excluding {case} due to npolynomcons'.format(case=case_name))
+            continue
+
+        nsignomcons = int(row[headings['nsignomcons']])
+        if nsignomcons < min_nsignomcons or nsignomcons > max_nsignomcons:
+            logger.debug('excluding {case} due to nsignomcons'.format(case=case_name))
+            continue
+
+        ngennlcons = int(row[headings['ngennlcons']])
+        if ngennlcons < min_ngennlcons or ngennlcons > max_ngennlcons:
+            logger.debug('excluding {case} due to ngennlcons'.format(case=case_name))
+            continue
+
+        njacobiannz = int(row[headings['njacobiannz']])
+        if njacobiannz < min_njacobiannz or njacobiannz > max_njacobiannz:
+            logger.debug('excluding {case} due to njacobiannz'.format(case=case_name))
+            continue
+
+        njacobiannlnz = int(row[headings['njacobiannlnz']])
+        if njacobiannlnz < min_njacobiannlnz or njacobiannlnz > max_njacobiannlnz:
+            logger.debug('excluding {case} due to njacobiannlnz'.format(case=case_name))
+            continue
+
+        nlaghessiannz = int(row[headings['nlaghessiannz']])
+        if nlaghessiannz < min_nlaghessiannz or nlaghessiannz > max_nlaghessiannz:
+            logger.debug('excluding {case} due to nlaghessiannz'.format(case=case_name))
+            continue
+
+        nlaghessiandiagnz = int(row[headings['nlaghessiandiagnz']])
+        if nlaghessiandiagnz < min_nlaghessiandiagnz or nlaghessiandiagnz > max_nlaghessiandiagnz:
+            logger.debug('excluding {case} due to nlaghessiandiagnz'.format(case=case_name))
+            continue
+
+        nsemi = int(row[headings['nsemi']])
+        if nsemi < min_nsemi or nsemi > max_nsemi:
+            logger.debug('excluding {case} due to nsemi'.format(case=case_name))
+            continue
+
+        nnlsemi = int(row[headings['nnlsemi']])
+        if nnlsemi < min_nnlsemi or nnlsemi > max_nnlsemi:
+            logger.debug('excluding {case} due to nnlsemi'.format(case=case_name))
+            continue
+
+        nsos1 = int(row[headings['nsos1']])
+        if nsos1 < min_nsos1 or nsos1 > max_nsos1:
+            logger.debug('excluding {case} due to nsos1'.format(case=case_name))
+            continue
+
+        nsos2 = int(row[headings['nsos2']])
+        if nsos2 < min_nsos2 or nsos2 > max_nsos2:
+            logger.debug('excluding {case} due to nsos2'.format(case=case_name))
+            continue
+
+        cases.append(case_name)
+
+    return cases
```

Comment:
There has *got* to be an easier way to do this.  What about only taking 3 arguments: min, max, and acceptable.  Each would be a dict.  Missing entries in the dict would be assumed to be non-binding.  All dict entries would be validated against the column headers from the CSV.

Reply:
I do not want to use a dict because the explicit keyword arguments directly document which column headings in the csv are utilized/filtered against. I think I did the next best thing - I created some helper functions.